### DISCUSSION TITLE
Need to disable parallel_tool_calling in .net too

### DISCUSSION THREAD

**Discussion by kenvanhyning** [microsoft/semantic-kernel#9661] [General]
My team is working on a .net SK application and we want to use gpt-4o mini but we're seeing the 'string does not match pattern' error regularly with that model. 
 I saw at the end of the thread on this issue:  https://github.com/microsoft/semantic-kernel/issues/8292
 
 that support for .net to disable parallel tool calling is coming soon.  I didn't see any issue for it though.  Any idea on when it will be available?

**Comment by moonbox3** [comment]
Hi @kenvanhyning, thanks for your question. The ability to toggle `parallel_tool_calls` in .Net is available. In the following .Net concept sample, it shows the various ways to configure function calling options: https://github.com/microsoft/semantic-kernel/blob/7579b243f4f428d984e9d99300dde7256eb85a06/dotnet/samples/Concepts/FunctionCalling/FunctionCalling.cs#L478.

You'll want to do the following:

```c#
FunctionChoiceBehaviorOptions options = new() { AllowParallelCalls = false };

OpenAIPromptExecutionSettings settings = new() { FunctionChoiceBehavior = FunctionChoiceBehavior.Auto(options: options) };
```

**Comment by kenvanhyning** [comment]
Great - thank you!  I'll try this out now.  Appreciate the quick response.
